apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/PathExpectationSamples.kt

```diff
@@ -29,12 +28,362 @@ class PathExpectationSamples {
+                toEqual("txtt")
+            }
+        }
+    }
+
+    @Test
+    fun testFileNameFeature() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).fileName.toEndWith(TEST_DIR).notToStartWith(INVALID_DIR).notToBeBlank()
+
+        fails {
+            expect(dir).fileName.toEndWith(INVALID_DIR).toStartWith(TEST_DIR)
+        }
+    }
+
+    @Test
+    fun testFileName() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).fileName {
+            toEndWith(TEST_DIR)
+            notToStartWith(INVALID_DIR)
+            notToBeBlank()
+        }
+
+        fails {
+            expect(dir).fileName {
+                toEndWith(INVALID_DIR)
+                toStartWith(TEST_DIR)
+            }
+        }
+    }
+
+    @Test
+    fun testFileNameWithoutExtensionFeature() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).fileNameWithoutExtension.notToBeEmpty().toEqual(TEST_DIR)
+
+        fails {
+            expect(dir).fileNameWithoutExtension.toBeEmpty().notToEqual(TEST_DIR)
+        }
+    }
+
+    @Test
+    fun testFileNameWithoutExtension() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).fileNameWithoutExtension {
+            notToBeEmpty()
+            toEqual(TEST_DIR)
+        }
+
+        fails {
+            expect(dir).fileNameWithoutExtension {
+                toBeEmpty()
+                notToEqual(TEST_DIR)
+            }
+        }
+    }
+
+    @Test
+    fun testParentFeature() {
+        val dir = tempDir.newDirectory("${TEST_DIR}_1")
+        val dir2 = tempDir.newDirectory("${TEST_DIR}_2")
+
+        expect(dir).parent.toEqual(dir2.parent)
+
+        fails {
+            expect(dir).parent.toEqual(dir2.newDirectory(TEST_DIR))
+        }
+    }
+
+    @Test
+    fun testParent() {
+        val dir = tempDir.newDirectory("${TEST_DIR}_1")
+        val dir2 = tempDir.newDirectory("${TEST_DIR}_2")
+
+        expect(dir).parent {
+            toEqual(dir2.parent)
+        }
+
+        fails {
+            expect(dir).parent {
+                toEqual(dir2.newDirectory(TEST_DIR))
+            }
+        }
+    }
+
+    @Test
+    fun testResolveFeature() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+        val fileInDir = dir.newFile("$TEST_FILE.txt")
+
+        expect(dir).resolve("$TEST_FILE.txt").toEqual(fileInDir)
+
+        fails {
+            expect(dir).resolve("$TEST_FILE.ttt").toEqual(fileInDir)
+        }
+    }
+
+    @Test
+    fun testResolve() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+        val fileInDir = dir.newFile("$TEST_FILE.txt")
+
+        expect(dir).resolve("$TEST_FILE.txt") {
+            toEqual(fileInDir)
+            toExist()
+        }
+
+        fails {
+            expect(dir).resolve("$TEST_FILE.ttt") {
+                toEqual(fileInDir)
+            }
+        }
+    }
+
+    private companion object {
+        private const val TEST_DIR = "test_dir"
```

Comment:
IMO sample files build an exception to the rule of reducing code duplication. Here I would not introduce these constants because the user which looks at the KDOC does not see that they stand for which makes understanding the samples harder. Please inline all those constants